rasa/core/nlg/generator.py

```diff
@@ -22,7 +22,7 @@ async def generate(
         """Generate a response for the requested template.
```

Comment:
@wochinge: is response == utterance == template? https://rasa.com/docs/rasa/glossary#response--template--utterance. Then what does this comment mean?

Reply:
We really need a good domain model 🙈 

I'd say `response == utterance`. In this case `template` means the action name of the response. So the `template` / name of the action is somewhat of a placeholder for the actual response. Means in this context distinguishing between `template` and `response` makes sense to me. We could also call it something like `response_name`?

Consider a domain like this:

```yaml
responses:
  utter_greet: # that's the "template name"
  - text: "hi". # actual response. Rasa will pick randomly between this one and the next one
  - text: "how are you?"  # another actual response
```

To fully confuse you: When you have the domain from above, Rasa Open Source will internally convert it to this:

```yaml
actions:
- utter_greet # utter greet is just an action like everything else. It's an action which sends back hardcoded responses.

responses:
  utter_greet: # that's the "template name"
  - text: "hi". # actual response. Rasa will pick randomly between this one and the next one
  - text: "how are you?"  # another actual response
```


Does that make sense? 🙈